src/version/client.go

```diff
@@ -21,18 +21,57 @@ var (
 	// AdditionalVersion is set here, the go linker will not overwrite it.
 	AdditionalVersion string
 
+	B = getBuildInfo()
+
 	// Version is the current version for pachyderm.
 	Version = &pb.Version{
-		Major:      uint32(MajorVersion),
-		Minor:      uint32(MinorVersion),
-		Micro:      uint32(MicroVersion),
-		Additional: AdditionalVersion,
+		Major:           uint32(MajorVersion),
+		Minor:           uint32(MinorVersion),
+		Micro:           uint32(MicroVersion),
+		Additional:      AdditionalVersion,
+		GitCommit:       B.gitCommit,
```

Comment:
if `debug.ReadBuildInfo()` returns error and b` `is empty won't this lead to `panic`? i guess `debug.ReadBuildInfo() `would be unlikely to have an issue?

Reply:
the default value for all of the fields in buildInfo is "", because that's the default for a string in go, not nil.  So there's no chance of a nil pointer here, since we're always returning a struct of buildinfo, it's just filled with the default values when it's empty in this case.